report_builder/models.py

```diff
@@ -138,6 +141,153 @@ def get_good_display_fields(self):
                 bad_display_fields.append(display_field)
         return display_fields.exclude(id__in=[o.id for o in bad_display_fields])
 
+    def report_to_list(self, queryset=None, user=None, preview=False):
```

Comment:
Also this is a pretty long method. It would be a lot more readable if it could be broken up.